canvasxyz/canvas
# WIP: add one click login

## Description
This currently works for chat-webpack.

TO DO:
- [ ] Update chat-next
- [ ] This hook relies on wagmi, it could possibly live in the new signers/ethereum package. It probably shouldn't live in hooks unless we can make it chain-agnostic.
- [ ] Write up what QA I did
- [ ] Document the actual state transitions

Here's how I QA'd it:
single click login tests:
- click "Connect" having not logged in before, unlock wallet, approve session signing, happy path etc
- click "Connect" having not logged in before, don't unlock wallet
- click "Connect" having not logged in before, unlock wallet, reject session signing
    - should disconnect, then try logging in again
- click "Connect", then refresh page
    - should log in automatically
- click "Connect", then delete CANVAS_SESSION key, then refresh
    - should be disconnected
    - should be able to log in again
- click "Connect", then lock wallet
    - should automatically disconnect
    - should be able to log back in


## How has this been tested?
- [ ] Tested with chat-next (including login, all signers, and exchanging messages)
- [x] Tested with chat-webpack (including login, all signers, and exchanging messages)
